packages/dcos-integration-test/extra/test_legacy_user_management.py

```diff
@@ -188,8 +189,55 @@ def test_dynamic_ui_config(dcos_api_session):
     assert 'uiConfiguration' in data
 
 
-def test_dcos_add_user_script_exists_oss():
+def test_dcos_add_user(dcos_api_session):
     """
-    dcos_add_user.py script exists in /opt/mesosphere/bin directory.
+    dcos_add_user.py script adds a user to IAM using the
+    script dcos_add_user.py.
     """
-    assert os.path.isfile("/opt/mesosphere/bin/dcos_add_user.py")
+
+    email_address = uuid.uuid4().hex + '@example.com'
+    cli = dcos_cli.DcosCli('')
+    command = ['python', '/opt/mesosphere/bin/dcos_add_user.py', email_address]
+    cli.exec_command(command)
+
+    r = dcos_api_session.get('/acs/api/v1/users')
+    r.raise_for_status()
+    expected_user_data = {
+        "uid": email_address,
+        "description": "",
+        "url": "/acs/api/v1/users/" + email_address,
+        "is_remote": True,
+        "is_service": False,
+        "provider_type": "oidc",
+        "provider_id": "https://dcos.auth0.com/"
+    }
+
+    delete_user(dcos_api_session, email_address)
+
+    assert expected_user_data in r.json()['array']
+
+
+def test_check_message_on_adding_user_twice(dcos_api_session):
+    """
+    Check that the correct message is emitted on adding the
+    same user for the second time.
+    """
+
+    email_address = uuid.uuid4().hex + '@example.com'
+    cli = dcos_cli.DcosCli('')
+    command = ['python', '/opt/mesosphere/bin/dcos_add_user.py', email_address]
+    stdout, stderr = cli.exec_command(command)
+
+    expected_output = '[INFO] Created IAM user `' + email_address + '`\n'
+    assert '' == stdout
+    assert expected_output == stderr
+
+    # Adding the same user second time
+    stdout, stderr = cli.exec_command(command)
+
+    expected_error = '[INFO] User `' + email_address + '` already exists\n'
+
+    delete_user(dcos_api_session, email_address)
+
+    assert expected_error == stderr
+    assert '' == stdout
```

Comment:
This result implies an exit status of `0`. Should it be `1` if the user exists?

Reply:
It does have an exit status of `0`.
That isn't just implied, it is tested for - else `exec_command` would raise a `subprocess.CalledProcessError`.

I would like to not change this here.

Would you like me to make an issue to change this? Or just leave it?

Reply:
Leave it. It is kind of a success in an idempotent way. Although it is also something an admin might want to flag up.